### PR TITLE
EVG-2485 create task group model

### PR DESCRIPTION
**Pull request by john-m-liu**
Half the code changes are renaming things so you can ignore anything that's not model/project_*.go

### CODE REVIEW

**Comment by tychoish** [comment]
A few things/questions: 
- lets get the tests to pass.
- we should make sure that we add support for validation (like no negative/0 maxhosts, at least one task in a command group. etc.) 
- we should also maybe remove pre/post and denormalize that content into the task groups?
- I'm also interested/worried about the way this interacts with the next-task endpoint. is there a way to have a non-task-group aware agent handle these tasks. If not, we can have a new-new-next-task endpoint to handle these cases, but we should plan for that soon.

File: model/project_parser_test.go [hunk 1]
```diff
@@ -782,3 +783,239 @@ tasks:
 	assert.Equal("execTask2", proj.BuildVariants[0].DisplayTasks[1].ExecutionTasks[0])
 	assert.Equal("execTask4", proj.BuildVariants[0].DisplayTasks[1].ExecutionTasks[1])
 }
+
+func TestTaskGroupValidation(t *testing.T) {
```
Comment (bsamek) [inline model/project_parser_test.go]: Add a null case, i.e., something that includes no task groups. I think it's unlikely that this case will error, but I think it's worth verifying that validate does what you expect for the no task group case.

Comment (john-m-liu) [reply model/project_parser_test.go]: There are other tests in this package that make sure project yml parses without task groups, so I don't think it's necessary to add one here

**Comment by john-m-liu** [comment]
Add more validation and fixed tests. I think the pre/post and backwards compatibility with agents should be handle by the agent/API respectively rather than in the data model

File: model/project_parser_test.go (hunk 1 above)
Comment (bsamek) [reply model/project_parser_test.go]: sgtm

**Pull request merged**